### PR TITLE
feat: implement QUST parsing

### PR DESCRIPTION
**Pull request by ZikkeyLS**

### CODE REVIEW

**Comment by Rolothe5th** [comment]
sweet wish ya luck with this btw!

File: skymp5-server/cpp/server_guest_lib/MpChangeForms.h [hunk 1]
```diff
@@ -105,6 +106,9 @@ class MpChangeFormREFR
   // Used for SetDisplayName (object reference)
   std::optional<std::string> displayName;
 
+  // Used for Quest (QUST) synchronization
+  std::optional<std::vector<Quest>> quests;
```
Comment (Pospelove) [inline skymp5-server/cpp/server_guest_lib/MpChangeForms.h:110]: vector<optional<Quest>> -> optional<vector<Quest>>

File: skymp5-server/cpp/server_guest_lib/Quest.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <nlohmann/json.hpp>
+#include <simdjson.h>
+#include <string>
+
+class Quest
+{
+public:
+  bool SetStage(uint32_t stageID) const;
```
Comment (Pospelove) [inline skymp5-server/cpp/server_guest_lib/Quest.h:9]: class -> struct

Functions must be papyrus functions, not Quest methods

File: skymp5-server/cpp/server_guest_lib/MpChangeForms.h (hunk 1 above)
Comment (ZikkeyLS) [reply skymp5-server/cpp/server_guest_lib/MpChangeForms.h:110]: 110 seems like you wrote to change optional<vector>, what is wrong?

Comment (ZikkeyLS) [reply skymp5-server/cpp/server_guest_lib/MpChangeForms.h:110]: std::optional<std::vector>

Comment (Pospelove) [reply skymp5-server/cpp/server_guest_lib/MpChangeForms.h:110]: sorry

**Review by Pospelove — APPROVED** [review]

**Pull request merged**